docs: revised windows taskbar page (#26120)
electron/electron: docs/tutorial/windows-taskbar.md

# Windows Taskbar

## Overview

Electron has APIs to configure the app's icon in the Windows taskbar. This API
supports both Windows-only features like [creation of a `JumpList`](#jumplist),
[custom thumbnails and toolbars](#thumbnail-toolbars),
[icon overlays](#icon-overlays-in-taskbar), and the so-called
["Flash Frame" effect](#flash-frame), and cross-platform features
like [recent documents][recent-documents] and
[application progress][progress-bar].

### JumpList

Windows allows apps to define a custom context menu that shows up when users

> right-click the app's icon in the task bar. That context menu is called

Reviewer: ```suggestion
right-click the app's icon in the taskbar. That context menu is called
```
spelling consistency

])
```

##### Clear tasks list

To clear your tasks list, you need to call `app.setUserTasks` with an empty
array in the `main.js` file.

```javascript
const { app } = require('electron')

app.setUserTasks([])
```

> NOTE: The user tasks will still be displayed even after closing your
application, so the icon and program path specified for a task should exist until your application is uninstalled.

[msdn-jumplist]: https://docs.microsoft.com/en-us/windows/win32/shell/taskbar-extensions#tasks

### Thumbnail Toolbars

> In Windows, you can add a thumbnail toolbar with specified buttons to a taskbar

Reviewer: ```suggestion
On Windows, you can add a thumbnail toolbar with specified buttons to a taskbar
```

win.setThumbarButtons([])
```

[msdn-thumbnail]: https://docs.microsoft.com/en-us/windows/win32/shell/taskbar-extensions#thumbnail-toolbars

### Icon Overlays in Taskbar

> In Windows, a taskbar button can use a small overlay to display application

Reviewer: ```suggestion
On Windows, a taskbar button can use a small overlay to display application
```

> network status, messenger status, or new mail. The user should not be
> presented with constantly changing overlays or animations.

![Overlay on taskbar button](https://i-msdn.sec.s-msft.com/dynimg/IC420441.png)

> NOTE: The screenshot above is an example of overlay on a taskbar button

To set the overlay icon for a window, you need to use the
[BrowserWindow.setOverlayIcon][setoverlayicon] API.

#### Example

Starting with a working application from the
[Quick Start Guide](quick-start.md), update the `main.js` file with the
following lines:

```javascript
const { BrowserWindow } = require('electron')

const win = new BrowserWindow()

win.setOverlayIcon('path/to/overlay.png', 'Description for overlay')
```

[msdn-icon-overlay]: https://docs.microsoft.com/en-us/windows/win32/shell/taskbar-extensions#icon-overlays

### Flash Frame

> In Windows, you can highlight the taskbar button to get the user's attention.

Reviewer: ```suggestion
On Windows, you can highlight the taskbar button to get the user's attention.
```

This is similar to bouncing the dock icon in macOS.

As quoted from [MSDN][msdn-flash-frame]:

> Typically, a window is flashed to inform the user that the window requires
> attention but that it does not currently have the keyboard focus.

To flash the BrowserWindow taskbar button, you need to use the
[BrowserWindow.flashFrame][flashframe] API.

#### Example

Starting with a working application from the
[Quick Start Guide](quick-start.md), update the `main.js` file with the
following lines:

```javascript
const { BrowserWindow } = require('electron')

const win = new BrowserWindow()

win.once('focus', () => win.flashFrame(false))
win.flashFrame(true)
```

> > NOTE: Don't forget to call the `flashFrame` method with `false` to turn off

Reviewer: ```suggestion
> NOTE: Don't forget to call `win.flashFrame(false)` to turn off
```

conciseness
